tests/rbd/trash_restore_same_name_image.py

```diff
@@ -0,0 +1,95 @@
+"""Verify there is no name collision with a trashed image.
+This module verifies trash operations
+
+Pre-requisites :
+
+We need atleast one client node with ceph-common package,
+conf and keyring files
+
+Test cases covered -
+
+CEPH-11393 - Delayed Deletion - Verify there is no name collision with a trashed image
+
+Test case flow -
+1. Create a Rep pool and an Image
+2. generate IO in images
+3. Move images to trash
+4. Create a same(trashed image) name image again.
+4. Restore the moved image in trash
+5. verify restore failes with message as
+'an image with the same name already exists,
+try again with with a different name'
+6. Repeat the above steps for EC-pool.
+"""
+
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def trash_restore_same_name_image(rbd, pool_type, **kw):
+    """
+    Run move image to trash and restore back the image from trash with same name
+        Args:
+        rbd: rbd object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: Test data
+    """
+    pool = kw["config"][pool_type]["pool"]
+    image = kw["config"][pool_type]["image"]
+
+    try:
+        # run ios on image and move image to trash
+        client = kw["ceph_cluster"].get_nodes(role="client")[0]
+        run_fio(image_name=image, pool_name=pool, client_node=client)
+        rbd.move_image_trash(pool, image)
+
+        # Create a image with same name as moved to trash
+        rbd.create_image(pool, image, "10G")
+
+        # Restore the image from trash
+        image_id = rbd.get_image_id(pool, image)
+        log.info(f"image id is {image_id}")
+        out, err = rbd.trash_restore(pool, image_id, all=True, check_ec=False)
+        if "an image with the same name already exists" in err:
+            log.info("As expected Restoration of image with same name is restricted")
+            return 0
```

Comment:
I don't see this log.info message in logs
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-omyph/Test_to_verify_image_restore_with_same_name_0.log

code was added after running tests ?

Reply:
latest log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-RISMX8